### PR TITLE
feat: use namespace annotations

### PR DESCRIPTION
**Pull request by kkujawa-sumo**
feat: use namespace annotations to include/exclude namespace from collection or set sourceCategory, sourceHost, and sourceName

### CODE REVIEW

**Review by sumo-drosiek — APPROVED** [review]
Nice! :guitar:

File: pkg/processor/sourceprocessor/source_category_filler_test.go [hunk 1]
```diff
@@ -66,6 +66,24 @@ func TestFillWithAnnotations(t *testing.T) {
 	assertAttribute(t, attrs, "_sourceCategory", "ABC#123asd#Prefix:sc#from#annot#ns#1#123asd")
 }
 
+// TODO: add more test cases
```
Comment (swiatekm-sumo) [inline pkg/processor/sourceprocessor/source_category_filler_test.go:69]: Are you planning on doing so in this PR?

**Review by swiatekm-sumo — COMMENTED** [review]
Code changes LGTM, some comments about documentation.

File: pkg/processor/sourceprocessor/source_category_filler_test.go (hunk 1 above)
Comment (kkujawa-sumo) [reply pkg/processor/sourceprocessor/source_category_filler_test.go:69]: No, I added tests but I forget to remove this comment :P

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**